src/System.Diagnostics.StackTrace/src/System/Diagnostics/StackTraceSymbols.CoreCLR.cs

```diff
@@ -0,0 +1,244 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+extern alias SRE; 
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Reflection.PortableExecutable;
+
+using Path = SRE::System.IO.Path;  
+using BitConverter = SRE::System.BitConverter;  
+
+namespace System.Diagnostics
+{
+    public class StackTraceSymbols : IDisposable
+    {
+        Dictionary<IntPtr, Tuple<MetadataReaderProvider, MetadataReader>> _readerCache;
+
+        /// <summary>
+        /// Create an instance of this class.
+        /// </summary>
+        public StackTraceSymbols()
+        {
+            _readerCache = new Dictionary<IntPtr, Tuple<MetadataReaderProvider, MetadataReader>>();
+        }
+
+        /// <summary>
+        /// Clean up any cached providers.
+        /// </summary>
+        void IDisposable.Dispose()
+        {
+            foreach (Tuple<MetadataReaderProvider, MetadataReader> tuple in _readerCache.Values)
+            {
+                tuple.Item1.Dispose();
+            }
+            _readerCache.Clear();
```

Comment:
Is there a reason to force the clearing here?  Is the fear someone will use the object after it has been disposed and things will go bad?